src/test/java/net/snowflake/client/config/SFConnectionConfigParserTest.java

```diff
@@ -73,37 +73,87 @@ public void testLoadSFConnectionConfigWithTokenFromFile()
     SnowflakeUtil.systemSetEnv(SNOWFLAKE_DEFAULT_CONNECTION_NAME_KEY, "default");
     File tokenFile = new File(Paths.get(tempPath.toString(), "token").toUri());
     prepareConnectionConfigurationTomlFile(
-        Collections.singletonMap("token_file_path", tokenFile.toString()), true);
+        Collections.singletonMap("token_file_path", tokenFile.toString()));
 
     ConnectionParameters data = SFConnectionConfigParser.buildConnectionParameters();
     assertNotNull(data);
     assertEquals(tokenFile.toString(), data.getParams().get("token_file_path"));
-    assertEquals("testToken", data.getParams().get("token"));
+  }
+
+  @Test
+  public void testThrowErrorWhenWrongPermissionsForConnectionConfigurationFile()
+      throws IOException {
+    SnowflakeUtil.systemSetEnv(SNOWFLAKE_HOME_KEY, tempPath.toString());
+    File tokenFile = new File(Paths.get(tempPath.toString(), "token").toUri());
+    prepareConnectionConfigurationTomlFile(
+        Collections.singletonMap("token_file_path", tokenFile.toString()), false, false);
+    assumeFalse(RunningNotOnLinuxMac.isNotRunningOnLinuxMac());
```

Comment:
perhaps it would be better to use this in  @ConditionalIgnoreRule.ConditionalIgnore annotation?

Reply:
I've tried and it wasn't work. I think we can stay with this solution.